### ISSUE TITLE
macOS: Cert verify failed: 8 (The certificate is not correctly signed by the trusted CA)

### ISSUE THREAD

**Issue by yuriy-chumak** [obsproject/obs-studio#9315]
### Operating System Info

macOS 13

### OBS Studio Version

Git

### OBS Studio Version (Other)

f7e6a5c8e7c9d9c955e295e06c46e83b3b60ef68

### OBS Studio Log URL

https://obsproject.com/logs/R3FaJcalBksZyqDe

### Expected Behavior

Streaming is started.

### Current Behavior

No streaming started. Error window appeared.

![image](https://github.com/obsproject/obs-studio/assets/2684656/99eafaef-fea4-4d11-adcf-09900ff412d5)

### Steps to Reproduce

1. checkout master (tested with almost master, the f7e6a5c8e7c9d9c955e295e06c46e83b3b60ef68 commit),
2. build, run,
4. select "YouTube - RTMPS", connect account,
5. press "start streaming" button, create broadcast,
6. watch error notification window.

### Anything else we should know?

Caused by f7ec81ca12e4408778d7c5eef5f1cd152cb5e6b4 (Revert "obs-outputs: Fix keychain API deprecation warnings on macOS") commit.

The build with `git revert f7ec81ca12e4408778d7c5eef5f1cd152cb5e6b4` works fine. Proper logs:
https://obsproject.com/logs/5NpR3mOYoVhn6q39

**Comment by PatTheMav** [comment]
I don't think that the change itself is the root cause because I have been able to stream to YouTube using RTMPS and HLS without reverting it. But I also get this issue seemingly at random, so it must be some other quirk we might be running into.

And for what it's worth I _also_ got the issue with the old implementation iirc.

**Comment by PatTheMav** [comment]
I just checked and was able to stream to YouTube just fine with RTMPS and HLS, so the code _does_ work. But as I mentioned before I also ran into the same error, so I still don't know what might trigger it.

**Comment by yuriy-chumak** [comment]
Just retested with latest master. Still confirming issue.

Hmmmmmmm. I'll try to find another one Mac to test on.

**Comment by RytoEX** [comment]
Could you please test with #9358 applied?

**Comment by yuriy-chumak** [comment]
> Could you please test with #9358 applied?

Tried. No, same issue.

Tested with:
![image](https://github.com/obsproject/obs-studio/assets/2684656/e458c2c1-e2d1-4827-a85c-7c2428540dac)

**Comment by PatTheMav** [comment]
Tried again to reproduce this today, but had no issue with YouTube RTMPS output. My next attempt will be after installing an macOS update or something to figure out if this is connected to Keychain availability or unlocking.

**Comment by yuriy-chumak** [comment]
Fixed with 121fb8b2fdc8fc8b73fc29493096129646b9eb61. Confirm.
